tests/generic_config_updater/test_aaa.py

```diff
@@ -0,0 +1,609 @@
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+AAA_CATEGORY = ["authentication", "authorization", "accounting"]
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname, tbinfo):
+    """
+    Setup/teardown fixture for each loopback interface test.
+    rollback to check if it goes back to starting config
+
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+
+
+def verify_aaa_sub_options(duthost, aaa_type, option, value):
+    """ Verify if AAA sub type's options match with expected value
+
+    Sample output:
+    admin@vlab-01:~/tacacs$ show aaa
+    AAA authentication login local (default)
+    AAA authentication failthrough False (default)
+    AAA authorization login local (default)
+    AAA accounting login disable (default)
+    """
+    output = duthost.shell('show aaa | grep -Po "AAA {} {} \K.*"'.format(aaa_type, option))
```

Comment:
>\K.*" [](http://example.com/codeflow?start=59&length=5)

Does this command work on something with `default` word?

```
admin@vlab-01:~$ show aaa
AAA authentication login local (default)
AAA authentication failthrough False (default)
```

Reply:
Yes.
```
admin@vlab-01:~$ show aaa  | grep -Po "AAA authentication login \K.*"
local (default)

```

Reply:
Then `verify_aaa_sub_options(duthost, aaa_type, "login", "local")` will assert. Is it unexpected?

Reply:
Before `verify_aaa_sub_options(duthost, aaa_type, "login", "local")`, we add the config command `aaa_add_init_config(duthost)`, so it will not contains the `default` keyword even though they mean the same thing. 
```
admin@vlab-01:~$ show aaa
AAA authentication login local (default)
AAA authentication failthrough False (default)
AAA authorization login local (default)
AAA accounting login disable (default)

admin@vlab-01:~$ sudo config aaa authentication login local
admin@vlab-01:~$ show aa
AAA authentication login local                     <=== no default keyword
AAA authentication failthrough False (default)
AAA authorization login local (default)
AAA accounting login disable (default)

```

Reply:
Changed to `get_aaa_sub_options_value` and do the verification afterwords.